paint.js

```diff
@@ -740,7 +740,7 @@ PaintCanvasMorph.prototype.floodfill = function (sourcepoint) {
         ctx = this.paper.getContext("2d"),
         img = ctx.getImageData(0, 0, width, height),
         data = img.data,
-        stack = [Math.round(sourcepoint.y) * width + sourcepoint.x],
+        stack = [Math.round(Math.round(sourcepoint.y) * width + sourcepoint.x)],
```

Comment:
I first tried `Math.round`ing the whole expression and it didn't work. The numbers were off by so much that it paints outside the shape. You absolutely need to round `y`.

`parseInt` instead of `Math.round` at the outer expression also does work, but what it actually does is truncate the float at the integer part, which would result in a pixel being off in some edge cases.